pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java

```diff
@@ -519,12 +519,19 @@ private Object applyAnswerPredicate(InterpreterContext ictx, String userGuid, St
             }
         } else {
             var answerDao = new AnswerCachedDao(ictx.getHandle());
-            Optional<Answer> compositeAnswer;
             if (instanceGuid == null) {
-                compositeAnswer = answerDao.findAnswerByLatestInstanceAndQuestionStableId(userGuid, studyId, stableId);
-            } else {
-                compositeAnswer = answerDao.findAnswerByInstanceGuidAndQuestionStableId(instanceGuid, stableId);
+                //try to find latest activity instance so that if composite answers exists can load child answers
+                instanceGuid = ictx.getHandle().attach(ActivityInstanceDao.class)
+                        .findLatestActivityInstanceGuidByUserStudyAndQuestionStableId(userGuid, studyId, stableId);
```

Comment:
I think finding latest activity instance using question stable id is a little weird. We have the `activityCode`. Can we use that instead? I think it'll make more sense that way.